src/views/PickupLocationModal.vue

```diff
@@ -0,0 +1,209 @@
+      }
+      
+      try {
+        const storeLookupResp = await FacilityService.getStores(payload)
+        if (storeLookupResp.status !== 200 || hasError(storeLookupResp) || !storeLookupResp.data.response.numFound) {
+          return [];
+        } 
+        return storeLookupResp.data.response.docs
+      } catch (error) {
+        console.error(error)
+      }
+    },
+
+    async getLocation() {
+      try {
+        const shipGroupLocationResp = await FacilityService.getLocation({
+          "json": {
+            "query": `postcode: ${this.shipGroup.shipTo.postalAddress.postalCode}`
+          }
+        })
+
+        if (shipGroupLocationResp.status !== 200 || hasError(shipGroupLocationResp) || !shipGroupLocationResp.data.response.numFound) {
+          return '';
+        }
+        return shipGroupLocationResp.data.response.docs[0].location
+      } catch (error) {
+        console.error(error)
+      }
+    },
+
+    async checkInventory(facilityIds: Array<string>) {
+      try {
+        const productInventoryResp = await StockService.checkInventory({
+          "filters": {
+            "productId": this.productIds,
+            "facilityId": facilityIds
+          },
+          "fieldsToSelect": ["atp", "facilityName", "facilityId"],
+        });
+
+        if (productInventoryResp.status !== 200 || hasError(productInventoryResp) || !productInventoryResp.data.count) {
+          return [];
+        }
+        return productInventoryResp.data.docs.filter((store: any) => store.atp > 0)
+      } catch (error) {
+        console.error(error)
+      }
+    },
+
+    async showStores() {
+      try {
+        let stores;
+        if (this.shipGroup.shipmentMethodTypeId === 'STOREPICKUP') {
+          // shipgroup is in brokering queue in this case so we do not
+          // have any facility hence, all the stores are fetched
+          stores = await this.getStores()
+        } else {
+          const location = await this.getLocation()
+          if (!location) return showToast(translate("Something went wrong while fetching stores"));
+          stores = await this.getStores(location)
+        }
+
+        if (!stores?.length) return showToast(translate("Something went wrong while fetching stores"));
+
+        const facilityIds = stores.map((store: any) => store.storeCode)
+        const storesWithInventory = await this.checkInventory(facilityIds)
+
+        if (!storesWithInventory?.length) return showToast(translate("Something went wrong while fetching stores"));
+        stores.map((storeData: any) => {
+          const inventoryDetails = storesWithInventory.find((store: any) => store.facilityId === storeData.storeCode);
+          if (inventoryDetails) this.nearbyStores.push({ ...inventoryDetails, distance: storeData.dist });
+        });
+      } catch (error) {
+        console.error(error)
+        showToast(translate("Something went wrong while fetching stores"));
+      }
+    },
+
+    updateShipmentStore() {
```

Comment:
Use term consistently, either store or facility
```suggestion
    updateFacility() {
```